src/providers/swap-router-provider.ts

```diff
@@ -0,0 +1,77 @@
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core';
+import { SwapRouter02__factory } from '../types/other';
+import { log } from '../util';
+import { IMulticallProvider } from './multicall-provider';
+
+type ApprovalTypes = {
+  approvalTokenIn: number;
```

Comment:
Lets use the enum from RouterSDK instead of number? https://github.com/Uniswap/router-sdk/blob/main/src/approveAndCall.ts#L7-L13

Hopefully its exported by the package

Reply:
making some changes to router-sdk to export it 😬